waspc/data/Generator/templates/react-app/src/index.js

```diff
@@ -34,7 +37,12 @@ async function render() {
   const queryClient = await queryClientInitialized
   ReactDOM.render(
     <QueryClientProvider client={queryClient}>
-      { router }
+      {=# rootComponent.isDefined =}
+      <{= rootComponent.importIdentifier =}>{router}</{= rootComponent.importIdentifier =}>
+      {=/ rootComponent.isDefined =}
+      {=^ rootComponent.isDefined =}
+      {router}
+      {=/ rootComponent.isDefined =}
```

Comment:
Nitpick: What about:

```suggestion
      {=# rootComponent.isDefined =}
      <{= rootComponent.importIdentifier =}>
      {=/ rootComponent.isDefined =}
        {router}
      {=# rootComponent.isDefined =}
      </{= rootComponent.importIdentifier =}>
      {=/ rootComponent.isDefined =}
```

is this maybe nicer / more composable?

Reply:
It will be easier to add new user defined wrappers, but will we have the need to add them in the future?

Your approach has extra whitespace in the output when not using `rootComponent` e.g. `   {router}` 😂

Reply:
Yeah I know but I gave up on formatting, we can't do it 100% anyway.

I think what I wrote is more explicit about "we are just wrapping it into this".

Reply:
Makes sense, I'll probably need to give up on output formatting as well 😢 

Maybe in the future, one of our build steps can be running everything through Prettier auto-format and then the users will get super nice output :D

Reply:
yes that is probably what we will want to do! Or we will give up on it completely. But I like the idea of prettier.

Reply:
Updated here: https://github.com/wasp-lang/wasp/pull/1009/commits/207806c7cedd0093bfe9682f4a201d82fff4edd2